tools/crates-validator/src/types/repo.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright 2023 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::constants::{RISC_ZERO_REPO_GIT, RISC_ZERO_REPO_PATH_BUILD, RISC_ZERO_REPO_PATH_ZKVM};
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "kebab-case")]
+#[serde(untagged)]
+pub enum Repo {
+    Tag(TagRepo),
+    Branch(BranchRepo),
+    Path(PathRepo),
+}
+
+impl Repo {
+    pub fn tag(tag: impl Into<String>) -> Self {
+        Self::Tag(TagRepo { tag: tag.into() })
+    }
+
+    pub fn branch(branch: impl Into<String>) -> Self {
+        Self::Branch(BranchRepo {
+            branch: branch.into(),
+        })
+    }
+
+    pub fn path(path: impl Into<String>) -> Self {
+        Self::Path(PathRepo { path: path.into() })
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, serde::Serialize, serde::Deserialize)]
+pub struct TagRepo {
```

Comment:
Is there a specific reason to have these be dedicated types vs just `String`?

Reply:
When parsing the configuration file, if we use structs instead of strings we get the correct parsing syntax for free since `serde` will match against the struct's inner fields.